src/main/webapp/dev/js/common/questionRubric.es6

```diff
@@ -1,15 +1,162 @@
+        // question and column should exist, isSwapLeft must be boolean, question should not have existing responses
+        return;
+    }
+
+    const numberOfRows = parseInt($(`#rubricNumRows-${questionNum}`).val(), 10);
+    let swapColIndex;
+    const swapCellAccessorStr = `#rubric-options-row-${questionNum} .rubricCol-${questionNum}-${colIndex}`;
+    const rubricCellSelector = `td[class*='rubricCol-${questionNum}']`;
+
+    if (isSwapLeft) {
+        if ($(swapCellAccessorStr).prev(rubricCellSelector).length !== 0) {
+            swapColIndex = $(swapCellAccessorStr).prev(rubricCellSelector).attr('data-col');
+        } else {
+            // trying to swap left most column to left
+            return;
+        }
+    } else if ($(swapCellAccessorStr).next(rubricCellSelector).length !== 0) {
+        swapColIndex = $(swapCellAccessorStr).next(rubricCellSelector).attr('data-col');
+    } else {
+        // trying to swap right most column to right
+        return;
+    }
+
+    // swap rubric choices
+    let $currentCell = $(`#rubricChoice-${questionNum}-${colIndex}`);
+    let $swapCell = $(`#rubricChoice-${questionNum}-${swapColIndex}`);
+    let temp = $currentCell.val();
+
+    $currentCell.val($swapCell.val());
+    $swapCell.val(temp);
+
+    // swap rubric weights
+    $currentCell = $(`#rubricWeight-${questionNum}-${colIndex}`);
+    $swapCell = $(`#rubricWeight-${questionNum}-${swapColIndex}`);
+    temp = $currentCell.val();
+    $currentCell.val($swapCell.val());
+    $swapCell.val(temp);
+
+    // swap options filled
+    for (let row = 0; row < numberOfRows; row += 1) {
+        $currentCell = $(`#rubricDesc-${questionNum}-${row}-${colIndex}`);
+        $swapCell = $(`#rubricDesc-${questionNum}-${row}-${swapColIndex}`);
+        temp = $currentCell.val();
+        $currentCell.val($swapCell.val());
+        $swapCell.val(temp);
+    }
+}
+
+function disableCornerMoveRubricColumnButtons(questionNum) {
+    const $optionColumns = $(`#rubric-options-row-${questionNum} td[class*='rubricCol-']`);
+
+    // check left button of leftmost column, should be disabled
+    const $leftmostCol = $optionColumns.first();
+    const leftmostColIndex = $leftmostCol.attr('data-col');
+    const $leftmostColLeftBtn = $leftmostCol.find(`#rubric-move-col-left-${questionNum}-${leftmostColIndex}`);
+
+    if (!$leftmostColLeftBtn.prop('disabled')) {
+        $leftmostColLeftBtn.prop('disabled', true);
+    }
+
+    // check right button of rightmost column, should be disabled
+    const $rightmostCol = $optionColumns.last();
+    const rightmostColIndex = $rightmostCol.attr('data-col');
+    const $rightmostColRightBtn = $rightmostCol.find(`#rubric-move-col-right-${questionNum}-${rightmostColIndex}`);
+
+    if (!$rightmostColRightBtn.prop('disabled')) {
+        $rightmostColRightBtn.prop('disabled', true);
+    }
+
+    // check the second last column right button, should be enabled
+    if ($optionColumns.length > 2) {
```

Comment:
Actually `disableCornerMoveRubricColumnButtons()` doesn't work that way, it has nothing to do with `$optionColumns.length < 2`. There are just 3 conditions to check -

- Rightmost button should be disabled.
- Leftmost button should be disabled.
- Move right button of 2nd last column should be enabled (when instructor adds a new column).

Any further checks would only be redundant. So in my method I simply wrote the code to check for these 3 conditions only.